octobot_tentacles_manager/constants.py

```diff
@@ -0,0 +1,159 @@
+TENTACLES_REQUIREMENTS_INSTALL_TEMP_DIR = "requirements_temp_tentacles"
+TENTACLES_ARCHIVE_ROOT = "reference_tentacles"
+
+# Tentacles folders
+TENTACLES_PATH = "tentacles"
+DEFAULT_BOT_PATH = "."
+TENTACLES_BACKTESTING_PATH = "Backtesting"
+TENTACLES_EVALUATOR_PATH = "Evaluator"
+TENTACLES_INTERFACES_PATH = "Interfaces"
+TENTACLES_NOTIFIERS_PATH = "Notifiers"
+TENTACLES_SERVICES_PATH = "Services"
+TENTACLES_SERVICES_FEEDS_PATH = "Services_feeds"
+TENTACLES_TRADING_PATH = "Trading"
+TENTACLES_WEBSOCKETS_PATH = "Websockets"
+
+# Tentacles sub-folders
+TENTACLE_MAX_SUB_FOLDERS_LEVEL = 3
+TENTACLES_BACKTESTING_COLLECTORS_PATH = "collectors"
+TENTACLES_BACKTESTING_CONVERTERS_PATH = "converters"
+TENTACLES_BACKTESTING_IMPORTERS_PATH = "importers"
+TENTACLES_BACKTESTING_THIRD_LEVEL_EXCHANGES_PATH = "exchanges"
+TENTACLES_BACKTESTING_THIRD_LEVEL_SOCIAL_PATH = "social"
+TENTACLES_EVALUATOR_REALTIME_PATH = "RealTime"
+TENTACLES_EVALUATOR_TA_PATH = "TA"
+TENTACLES_EVALUATOR_SOCIAL_PATH = "Social"
+TENTACLES_EVALUATOR_STRATEGIES_PATH = "Strategies"
+TENTACLES_EVALUATOR_UTIL_PATH = "Util"
+TENTACLES_TRADING_MODE_PATH = "Mode"
+TENTACLES_TRADING_EXCHANGE_PATH = "Exchange"
+TENTACLES_WEBSOCKETS_FEEDS_PATH = "feeds"
+
+# Tentacle local module folders
+TENTACLE_CONFIG = "config"
+TENTACLE_RESOURCES = "resources"
+TENTACLE_TESTS = "tests"
+
+# Tentacle creator
+TENTACLE_TEMPLATE_PATH = "templates"
+TENTACLE_TEMPLATE_DESCRIPTION = "description"
+TENTACLE_CONFIG_TEMPLATE_PRE_EXT = "_config"
+TENTACLE_TEMPLATE_EXT = ".template"
+TENTACLE_TEMPLATE_PRE_EXT = "_tentacle"
+
+# Tentacles that can be activated / deactivated
+ACTIVATABLE_TENTACLES = [
+    TENTACLES_EVALUATOR_REALTIME_PATH,
+    TENTACLES_EVALUATOR_SOCIAL_PATH,
+    TENTACLES_EVALUATOR_TA_PATH,
+    TENTACLES_EVALUATOR_STRATEGIES_PATH,
+    TENTACLES_TRADING_MODE_PATH,
+]
+
+# Tentacles architecture
+TENTACLES_FOLDERS_ARCH = {
+    TENTACLES_BACKTESTING_PATH: {
+        TENTACLES_BACKTESTING_COLLECTORS_PATH: [
+            TENTACLES_BACKTESTING_THIRD_LEVEL_EXCHANGES_PATH,
+            TENTACLES_BACKTESTING_THIRD_LEVEL_SOCIAL_PATH
+        ],
+        TENTACLES_BACKTESTING_CONVERTERS_PATH: [
+            TENTACLES_BACKTESTING_THIRD_LEVEL_EXCHANGES_PATH
+        ],
+        TENTACLES_BACKTESTING_IMPORTERS_PATH: [
+            TENTACLES_BACKTESTING_THIRD_LEVEL_EXCHANGES_PATH,
+            TENTACLES_BACKTESTING_THIRD_LEVEL_SOCIAL_PATH
+        ]
+    },
+    TENTACLES_EVALUATOR_PATH: [
+        TENTACLES_EVALUATOR_REALTIME_PATH,
+        TENTACLES_EVALUATOR_SOCIAL_PATH,
+        TENTACLES_EVALUATOR_TA_PATH,
+        TENTACLES_EVALUATOR_STRATEGIES_PATH,
+        TENTACLES_EVALUATOR_UTIL_PATH
+    ],
+    TENTACLES_INTERFACES_PATH: [],
+    TENTACLES_NOTIFIERS_PATH: [],
+    TENTACLES_SERVICES_PATH: [],
+    TENTACLES_SERVICES_FEEDS_PATH: [],
+    TENTACLES_TRADING_PATH: [
+        TENTACLES_TRADING_MODE_PATH,
+        TENTACLES_TRADING_EXCHANGE_PATH
+    ],
+    TENTACLES_WEBSOCKETS_PATH: [
```

Comment:
Should be removed after https://github.com/Drakkar-Software/OctoBot-Trading/pull/141